provider/github/review.go

```diff
@@ -207,13 +216,56 @@ func splitReviewRequest(review *github.PullRequestReviewRequest, n int) []*githu
 	return result
 }
 
+func newFooterTemplate(tpl string) (*template.Template, error) {
+	if tpl == "" {
+		return nil, ErrEmptyTemplate.New()
+	}
+
+	if strings.Index(tpl, "%s") >= 0 {
+		return nil, ErrOldTemplate.New(tpl)
+	}
+
+	template, err := template.New("footer").Parse(tpl)
+	if err != nil {
+		return nil, ErrParseTemplate.New(err)
+	}
+
+	return template.Option("missingkey=error"), nil
+}
+
 // addFootnote adds footnote link to text of a comment
-func addFootnote(text, tmpl, url string) string {
-	if text == "" || tmpl == "" || url == "" {
-		return text
+func addFootnote(
+	ctx context.Context,
+	comment string, tmpl *template.Template, analyzerConf *lookout.AnalyzerConfig,
+) string {
+	if comment == "" || tmpl == nil {
+		return comment
+	}
+
+	footer, err := getFootnote(tmpl, analyzerConf)
+	if err != nil {
+		ctxlog.Get(ctx).Warningf("footer could not be generated: %s", err)
+	}
+
+	return comment + footer
+}
+
+func getFootnote(tmpl *template.Template, analyzerConf *lookout.AnalyzerConfig) (string, error) {
+	if analyzerConf == nil || analyzerConf.Feedback == "" || analyzerConf.Name == "" {
```

Comment:
Isn't it more flexible to avoid checking for `Feedback` and `Name` in the configuration? I mean if would just execute the template and throw an error if some keys are missing, but I would not enforce having `Feedback` and `Name` in the configuration, for two reasons:
1. I might still want to use a static message for some reason (so no need to for extra keys in analyzer configuration),
2. I might want to use a custom footer whose keys are not `Feedback`  and `Name`. As an example:

````
providers:
  github:
    comment_footer: "_Comment made by analyzer '{{.Name}}' (version '{{.Version}}'), [report]({{.Feedback}}) or [email]({{.Email}})._"
analyzers:
  - name: Fancy Analyzer
    addr: ipv4://localhost:9930
    feedback: http://example.com/report-issue
    version: 1.0.0
    email: superdev@sourced.tech
  - name: Awesome Analyzer
    addr: ipv4://localhost:9931
```

Reply:
> it is not expected to pass custom fields to the footer, because `AnalyzerConfig` has already a fixed schema.

You're right, I missed that. Maybe by putting something in the `Settings`. Anyway

> For sure I think we can (should) improve the way the footer is configured from the analyzers, but imo this PR shouldn't be the place.

I agree that it is certainly not to be done in this PR and maybe wait for inputs from @src-d/product. 👍

Reply:
I agree with @se7entyse7en, if we are giving the flexibility of a template, it makes sense to allow templates that don't use any fields. Skipping `feedback` in the `yml` should be fine if it's not used by the template.

If we want to enforce a failure for an empty `feedback`, IMO it should be done with an exit on initialization.

Reply:
@se7entyse7en Template can already take values from `settings` key in `lookoutd` config.
```yml
providers:
  github:
    comment_footer: "_[open an issue]({{.Feedback}}) or send us an [email]({{.Settings.email}})._"
analyzers:
  - name: Fancy Analyzer
    feedback: http://example.com/report-issue
    settings:
        email: admin@example.org
```

Reply:
@carlosms if we don't validate `conf.Feedback`, it might be posted "wrong footers" when `conf.Feedback` is not provided;
let's consider this footer template:
```yml
    comment_footer: "If you have feedback, [open an issue]({{.Feedback}})"
```
if `conf.Feedback` is not provided for a certain analyzer, its footer will look as it follows:
```
If you have feedback, [open an issue]()
```
^ with a broken link.

Why? because the default value of the `conf.Feedback` is an empty string, so the template will be executed with no errors.

Should I parse the template to check if `{{.Feedback}}` placeholder is present? If we do so, then we can return the error if the `conf.Feedback` is empty only when it is required by the template.

imho :point_up: this is too much for this use case, and should be considered in a new issue about developing a fully customizable footer, but whatever you prefer.

Reply:
Ok! I missed that, that's cool! So we can actually avoid enforcing `feedback` in the conf right?

Reply:
As I said, we can ignore the absence of `conf.Feedback` when executing the template if:
- we don't care about broken links:
  ```
  If you have feedback, [open an issue]()
  ```
- we parse the template to ensure that `{{.Feedback}}` is not being used

Reply:
Yup, sorry I've just realized that I didn't refresh the page 😅, and I missed your comment. I never worked with templates so far, so I don't know how much effort would it take. But I'm ok in doing it in a separated issues. As you prefer.

Reply:
I would not choose to parse the template; I'd choose one of https://github.com/src-d/lookout/pull/495#issuecomment-461103028

Since 63fb1fd already addresses the suggestion made in this thread, I'll consider it as "resolved"; if you want to consider another approach, just say so :dancer: